### DISCUSSION TITLE
list of children sub pages in the index.md body

### DISCUSSION THREAD

**Discussion by StefanoCecere** [squidfunk/mkdocs-material#6088] [General]
hi
I couldn't find a native way to embed the list of all subpages
for example in page https://2050x.org/spazio/riferimenti/personaggi/ i would like just the list of subpages like in the navigation

is it possible without a custom plugin?

i know someone was asking the same for blog posts (to have a preview list in other pages)

thank you!

**Comment by squidfunk** [comment]
You could use a custom template for that page and iterate through all pages with:

``` jinja2
<ul>
  {% for page in nav.pages %}
    <li>{{ page.title }}</li>
  {% endfor %}
</ul>
```

**Comment by StefanoCecere** [comment]
i already wrote some templates to have custom tables.. great.

you confirm that MkDocs doesn't allow to inject "widgets" or shortcodes (like Hugo calls tham) directly in the .md file, right?
we have to go with an external page template...

**Reply by squidfunk** [reply]
You can add your own custom shortcodes with hooks, similar to how we do it in our docs:

https://github.com/squidfunk/mkdocs-material/blob/b77769009ece2326476d1a02d8fa04e501f73af0/src/overrides/hooks/shortcodes.py#L40-L70
